analytics_data_api/v0/documents.py

```diff
@@ -11,168 +11,60 @@ class RosterUpdate(Document):
-        }
-    )
-    location = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    year_of_birth = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    level_of_education = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    gender = Text(
-        fields={
-            'raw': Text(),
-        }
-    )
-    mailing_address = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    city = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    country = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    goals = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    enrollment_mode = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    cohort = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    segments = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )  # segments is an array/list of strings
-    problems_attempted = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    problems_completed = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    problem_attempts_per_completed = Float(
-        fields={
-            'raw': Float(),
-        }
-    )
+    course_id = Keyword()
+    user_id = Integer()
+    username = Keyword()
+    name = Keyword()
+    email = Keyword()
+    language = Keyword()
+    location = Keyword()
+    year_of_birth = Integer()
+    level_of_education = Keyword()
+    gender = Keyword()
+    mailing_address = Keyword()
+    city = Keyword()
+    country = Keyword()
+    goals = Keyword()
+    enrollment_mode = Keyword()
+    cohort = Keyword()
+    segments = Keyword()  # segments is an array/list of strings
+    problems_attempted = Integer()
+    problems_completed = Integer()
+    problem_attempts_per_completed = Float()
     # Useful for ordering problem_attempts_per_completed (because results can include null, which is
     # different from zero).  attempt_ratio_order is equal to the number of problem attempts if
     # problem_attempts_per_completed is > 1 and set to -problem_attempts if
     # problem_attempts_per_completed = 1.
-    attempt_ratio_order = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    discussion_contributions = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    videos_watched = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
+    attempt_ratio_order = Integer()
+    discussion_contributions = Integer()
+    videos_watched = Integer()
     enrollment_date = Date()
     last_updated = Date()
 
     class Index:
         name = settings.ELASTICSEARCH_LEARNERS_INDEX
-        settings = {
-            'number_of_shards': 1,
-            'number_of_replicas': 0
-        }
```

Comment:
Did you describe the setting globally or it was set by default? So why did you delete it?

Reply:
Document settings per index didn't set up in the api before, it is a reason why I removed them. Do I need to set up the on the global settings or in the every document index?

Reply:
I prefer to move it globally and do not declare the same settings for each index it has no sense. if we are wrong then our colleagues from OpenEdx will correct us.

Reply:
Thanks, add global setting for index